src/js/components/NetworkErrorNotice.tsx

```diff
@@ -39,7 +58,12 @@ export default function NetworkErrorNotice({error}: Props) {
     }
     tick()
 
-    return () => clearTimeout(id)
+    return () => {
+      clearTimeout(id)
+      if (statusRef.current === "retrying") {
```

Comment:
when the component exits, which may happen because the user has disregarded the notice and clicked somewhere else in the app which triggers a Notice.dismiss(), we check if the status is still set to "retrying" and if it is then we set it to "disconnected".

The idea here is that since they can't do anything with the conn while it is down, we give the options of retrying and keeping context, or if they choose to navigate away then they are already changing context and we can fall back on the <ConnectionError /> at that point.